src/Framebuffer.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "cuda_utils.h"
+
+class Framebuffer {
+public:
+    Framebuffer(Resolution res) {
+        resolution_ = res;
+        size_t pixel_count = res.width * res.height * sizeof(uint8_t) * 3;
+        checkCudaErrors(cudaMallocManaged((void **)&fb_ptr_, pixel_count));
+    }
+
+    void setResolution(Resolution res) {
+        if (res.width == resolution_.width && res.height == resolution_.height) {
+            return;
+        }
+        resolution_ = res;
+        size_t pixel_count = resolution_.width * resolution_.height * sizeof(uint8_t) * 3;
+        checkCudaErrors(cudaFree(fb_ptr_));
+        checkCudaErrors(cudaMallocManaged((void **)&fb_ptr_, pixel_count));
+    }
```

Comment:
I think this could be also solved with the observer design pattern - when Resolution object properties would change, they would notify the Framebuffer - however this approach is also ok.